client/finality-grandpa/src/communication/mod.rs

```diff
@@ -67,9 +67,23 @@ mod periodic;
 #[cfg(test)]
 pub(crate) mod tests;
 
-/// Name of the notifications protocol used by Grandpa. Must be registered towards the networking
-/// in order for Grandpa to properly function.
-pub const GRANDPA_PROTOCOL_NAME: &'static str = "/paritytech/grandpa/1";
+pub(crate) mod grandpa_protocol_name {
+	const NAME: &'static str = "/grandpa/1";
+	/// Old names for the notifications protocol, used for backward compatibility.
+	pub const LEGACY_NAMES: [&'static str; 1] = ["/paritytech/grandpa/1"];
+
+	/// Name of the notifications protocol used by Grandpa.
+	/// Must be registered towards the networking in order for Grandpa to properly function.
+	pub fn with_prefix(prefix: &str) -> String {
+		format!("{}{}", prefix, NAME)
+	}
+
+	#[cfg(test)]
+	/// Protocol name used in tests.
+	pub fn test_name() -> String {
+		format!("/test{}", NAME)
+	}
+}
```

Comment:
This feels overly complicated.

I know that the code of Grandpa is difficult to read, but that's because it is already too complicated. You might wonder why this `GRANDPA_PROTOCOL_NAME` is here, and the answer is that there's no reason. It's just one random person who one day decided to put this constant here, but there's no reason why it's here.
We shouldn't make Grandpa's code even more complicated just to maintain the completely arbitrary decision to put this constant here.
Unfortunately I'm not in the knows of the Grandpa code.

Reply:
I slightly simplified this, but the need for a `const NAME` is still there.
We want a single place to keep track of the _current_ protocol name as well as previous compatible ones (legacy called here).

`communications` mod looks like a good place because this is where the network and gossiping functionalities reside - and protocol name is part of these.